src/AuthenticatedApiClient/index.js

```diff
@@ -0,0 +1,223 @@
+import axios from 'axios';
+import {
+  csrfTokenProviderInterceptor,
+  jwtTokenProviderInterceptor,
+  processAxiosRequestErrorInterceptor,
+} from './axiosInterceptors';
+import { logFrontendAuthError } from './utils';
+import getJwtToken from './getJwtToken';
+
+let authenticatedApiClient = null;
+let config = null;
+
+function configure(incomingConfig) {
+  [
+    'appBaseUrl',
+    'loginUrl',
+    'logoutUrl',
+    // 'handleEmptyAccessToken', // optional
+    'loggingService',
+    'refreshAccessTokenEndpoint',
+    'accessTokenCookieName',
+    'csrfTokenApiPath',
+  ].forEach((key) => {
+    if (incomingConfig[key] === undefined) {
+      throw new Error(`Invalid configuration supplied to frontend auth. ${key} is required.`);
+    }
+  });
+
+  // validate the logging service
+  [
+    'logInfo',
+    'logError',
+  ].forEach((key) => {
+    if (incomingConfig.loggingService[key] === undefined) {
+      throw new Error(`Invalid configuration supplied to frontend auth. loggingService.${key} must be a function.`);
+    }
+  });
+
+  config = incomingConfig;
+}
+
+function getConfig(property) {
```

Comment:
I was looking at usage of this down below, and realized I thought of a small refactoring that I thought would be worthwhile... if `getConfig()` didn't take any parameters and just returned the whole config object, 1) you could reuse the config object you get instead of calling getConfig() over and over, and 2) the syntax for accessing something would be `getConfig().whatever` instead of `getConfig('whatever')` which feels a bit better.  IDE's might actually be able to click through, etc.

Reply:
It'd also be closer to what we're doing with accessing things elsewhere/feels a bit more canonical.

Reply:
Updated